libraries/commonIO/src/main/java/org/eclipse/sw360/commonIO/AttachmentFrontendUtils.java

```diff
@@ -40,6 +41,11 @@
     // TODO add Config class and DI
     private final Duration downloadTimeout = Duration.durationOf(30, TimeUnit.SECONDS);
 
+    protected final ThreadLocal<AttachmentService.Iface> attchmntClient = ThreadLocal.<AttachmentService.Iface>withInitial(
```

Comment:
I think you can delete the line
`protected final AttachmentService.Iface client;`
and cleanup the constructors because we don't need one with ThriftClients as parameter.